### ISSUE TITLE
Autocomplete for object destructuring in arrow function is bugged

### ISSUE THREAD

**Issue by senyaak** [TypeStrong/atom-typescript#1528]
For example we have following code 

```ts
([{a:1},{a:2},{a:3}] as {a: number}[]).forEach(({})=>null);
```
If you place pointer between '{' and '}' and then hit alt+space - autocomplete breaks the function.
What I expect
```ts
([{a:1},{a:2},{a:3}] as {a: number}[]).forEach(({a})=>null);
```
What I get 
```ts
([{a:1},{a:2},{a:3}] as {a: number}[]).forEacha})=>null);
```

PS 
Atom 1.41.0
Atom-Typescript 13.4.0

**Comment by lierdakil** [comment]
Thanks for reporting! Should be fixed in v13.4.1.

**Comment by senyaak** [comment]
Thank you for quick fix! 
But unfortunately there some bug in fix. Console says:
`Uncaught (in promise) Error: Cannot find module './misc'`

**Comment by lierdakil** [comment]
Huh. Ouch. Yup, sorry, that was my bad. Hopefully fixed in v13.4.3

**Comment by lierdakil** [comment]
Augh. Sorry, found another issue with the new implementation, when prefix with mismatched case wouldn't be replaced. v13.4.4 should be good.
